### PR TITLE
IEP-1226: added filter for python3 in dialog

### PR DESCRIPTION
**Pull request by alirana01**
## Description

Python selection dialog was not showing python3 due to missing filter

Fixes # ([IEP-1226](https://jira.espressif.com:8443/browse/IEP-1226))

## Type of change

Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)

## How has this been tested?

Use esp-idf manager view to add esp-idf and see if in the dialog the python3 filter is visible now

**Test Configuration**:
* ESP-IDF Version: any
* OS (Windows,Linux and macOS): all

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
  - Added support for parsing and handling Windows-specific file attributes, including reparse tags.
  - Enhanced file operations to determine file existence and directory status on Windows.

- **Improvements**
  - Updated the `isExecutable` method to better handle file existence checks based on the platform.
  - Improved Python file selection in the `BrowseButtonSelectionAdapter` class for better cross-platform compatibility.

### CODE REVIEW

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 

Tested under: 
OS - Windows 10 
ESP-IDF: v5.2.1 

Is it possible to combine both filters into 1 filter without any manual manipulations ? 
image

Because still, to display python3.exe I have to switch filter manually. The same works for PATH. I have to open "Browse" first -> then switch filter to python3.exe -> then close "Browse" -> and just then my path to python3 will be recognize by validator.

**Comment by alirana01** [comment]
> @alirana01 hi !
> 
> Tested under: OS - Windows 10 ESP-IDF: v5.2.1
> 
> Is it possible to combine both filters into 1 filter without any manual manipulations ? ![image](https://private-user-

I have pushed the changes but beware that now all python*.exe will be shown as the FileDialog can only use simple wildcards and does not support full regex

**Review by alirana01 — COMMENTED** [review]
Self Review @kolipakakondal @sigmaaa kindly review as well these changes will affect more areas as well since there is now some change in SystemExecutableFinder as well

File: bundles/com.espressif.idf.core/src/com/espressif/idf/core/SystemExecutableFinder.java [hunk 1]
```diff
@@ -109,8 +111,17 @@ protected boolean isPlatformWindows()
...
+		
+		if (isPlatformWindows() && !file.exists())
+		{
+			if (WinNativeFileTagOperations.fileExists(file))
+			{
+				return IDFUtil.isReparseTag(file);
+			}
+		}
+		
+
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.core/src/com/espressif/idf/core/SystemExecutableFinder.java:123]: coderabbit makes a good point here

**Review by sigmaaa — APPROVED** [review]
LGTM

**Comment by AndriiFilippov** [comment]
@alirana01 hi ! 

Tested under: 
OS - Windows 10 
ESP-IDF: v5.2.1 
Python: 3.11 / 3.12 / python3 

Filter works good now, able to see both python.exe / python3.exe without any manual filter switching 👍 
On Windows 10 everything works like it was ✔️ 

LGTM 👍

**Review by AndriiFilippov — APPROVED** [review]

**Pull request merged**